packages/store/addon/-private/system/references/reference.ts

```diff
@@ -61,6 +61,10 @@ abstract class Reference {
     }
   }
 
+  get recordData() {
+    return internalModelForReference(this)?._recordData;
```

Comment:
Can you get the identifier here as we for when we do have it and use it to lookup the record data

Reply:
Done!